lib/AST/ASTImporter.cpp

```diff
@@ -1641,13 +1691,16 @@ Decl *ASTNodeImporter::VisitStaticAssertDecl(StaticAssertDecl *D) {
   if (!ToMsg && FromMsg)
     return nullptr;
 
-  StaticAssertDecl *ToD = StaticAssertDecl::Create(
-        Importer.getToContext(), DC, Loc, AssertExpr, ToMsg,
-        Importer.Import(D->getRParenLoc()), D->isFailed());
+  bool NewlyCreated;
+  StaticAssertDecl *ToD;
+  std::tie(ToD, NewlyCreated) = CreateDecl<StaticAssertDecl>(
+      D, Importer.getToContext(), DC, Loc, AssertExpr, ToMsg,
+      Importer.Import(D->getRParenLoc()), D->isFailed());
+  if (!NewlyCreated)
+    return ToD;
```

Comment:
In case we end up returning here. we already created a StringLiteral and imported Expressions, Locations. None of them will be used. I wonder if this is a good idea.

Reply:
The `ASTNodeImporter` should work in a way that if something is imported once a next import of the same object returns the already imported object. For non-Decl this may not be true and if yes that can be problem.

Reply:
If we end up here that means that the same `StaticAssertDecl` had been imported already, which implies that the related `StringLiteral` and `AssertExpr` had been imported as well previously. Thus no new nodes will be created in this call.